src/pages/ProbeHybridisationQC.tsx

```diff
@@ -0,0 +1,391 @@
+          labwareSampleComments.comments.push(
+            ...mapSampleAddressCommentIds(address, Number(sampleId), labwareForm.sampleAddressComments[addressSampleId])
+          );
+        }
+      }
+    }
+    completionRequest.labware.push(labwareSampleComments);
+  });
+  return completionRequest;
+};
+
+const formInitialValues: ProbeHybridisationQCFormValues = {
+  workNumber: '',
+  labwares: {
+    '': {
+      globalComments: [],
+      completionDateTime: '',
+      sampleAddressComments: { '': [] }
+    }
+  }
+};
+
+export default function ProbeHybridisationQC({ comments }: SectionComments) {
+  const stanCore = useContext(StanCoreContext);
+
+  const labwares = useCollection<LabwareFieldsFragment>({
+    getKey: (item) => item.barcode
+  });
+
+  const onAddLabware = useCallback(
+    (labware: LabwareFieldsFragment) => {
+      labwares.append(labware);
+    },
+    [labwares]
+  );
+
+  const onRemoveLabware = useCallback(
+    (labware: LabwareFieldsFragment) => {
+      labwares.remove(labware.barcode);
+    },
+    [labwares]
+  );
+
+  const removeLabwareFromForm = useCallback(
+    (removeLabware, barcode: string, values: ProbeHybridisationQCFormValues) => {
+      removeLabware(barcode);
+      delete values.labwares[barcode];
+    },
+    []
+  );
+
+  const formSubmitMachine = React.useMemo(() => {
+    return createFormMachine<CompletionRequest, RecordCompletionMutation>().withConfig({
+      services: {
+        submitForm: (ctx, e) => {
+          if (e.type !== 'SUBMIT_FORM') return Promise.reject();
+          return stanCore.RecordCompletion({
+            request: e.values
+          });
+        }
+      }
+    });
+  }, [stanCore]);
+  const [currentForm, sendForm] = useMachine(() => formSubmitMachine);
+
+  const { serverError, submissionResult } = currentForm.context;
+
+  const validationWorkNumber = Yup.object().shape({
+    workNumber: Yup.string().required('SGP Number is required')
+  });
+
+  const submitForm = async (values: CompletionRequest) => sendForm({ type: 'SUBMIT_FORM', values });
+  const convertValuesAndSubmit = async (formValues: ProbeHybridisationQCFormValues) => {
+    const values: CompletionRequest = convertProbeHybridisationQCFormValuesToCompletionRequest(
+      formValues,
+      labwares.items
+    );
+    await submitForm(values);
+  };
+
+  const mapCommentOptionsToValues = useCallback((options: OptionType[]) => {
+    return options.map((v) => v.value);
+  }, []);
+
+  const updateSectionCommentsFromGlobal = useCallback(
+    (options: OptionType[], labware: LabwareFieldsFragment, values, setFieldValue) => {
+      labware.slots.forEach((slot) => {
+        slot.samples.forEach((sample) => {
+          const sampleAddressId = `${slot.address}-${sample.id}`;
+          const oldSelected = values.labwares[labware.barcode]?.sampleAddressComments?.[sampleAddressId] || [];
+          const updatedSelected = [...oldSelected, ...mapCommentOptionsToValues(options)];
```

Comment:
Why we need oldSelected and updateSelected?

Reply:
to cover the use case; where the user selects some options for a specific section and then adds other comments from the global comment

Reply:
Ohh, is it supposed to behave like that or is it to override everything by comments  from 'apply all' drop down when they change some value in it ? It is better to clarify with Katy.